File: scrimage-core/src/main/scala/com/sksamuel/scrimage/canvas/drawable/RoundedRect.scala
package com.sksamuel.scrimage.canvas.drawable

import java.awt.Graphics2D

import com.sksamuel.scrimage.canvas.{Drawable, GraphicsContext}

case class RoundedRect(x: Int,
                       y: Int,
                       width: Int,
                       height: Int,
                       arcWidth: Int,
                       arcHeight: Int,
                       context: GraphicsContext) extends Drawable {

  def draw(g: Graphics2D): Unit = g.drawRoundRect(x, y, width, height, arcWidth, arcHeight)

  def fill: FilledRoundedRect = FilledRoundedRect(x, y, width, height, arcWidth, arcHeight, context)

  def squared = Rect(x, y, width, height, context)
}

File: scrimage-core/src/main/java/com/sksamuel/scrimage/canvas/drawables/RoundedRect.java
package com.sksamuel.scrimage.canvas.drawables;

import com.sksamuel.scrimage.canvas.Drawable;
import com.sksamuel.scrimage.canvas.GraphicsContext;
import com.sksamuel.scrimage.canvas.RichGraphics2D;

public class RoundedRect implements Drawable {

    private final int x;
    private final int y;
    private final int width;
    private final int height;
    private final int arcWidth;
    private final int arcHeight;
    private final GraphicsContext context;

    public RoundedRect(int x, int y, int width, int height, int arcWidth, int arcHeight, GraphicsContext context) {
        this.x = x;
        this.y = y;
        this.width = width;
        this.height = height;
        this.arcWidth = arcWidth;
        this.arcHeight = arcHeight;
        this.context = context;
    }

    @Override
    public void draw(RichGraphics2D g) {
        g.drawRoundRect(x, y, width, height, arcWidth, arcHeight);
    }

    public FilledRoundedRect toFilled() {
        return new FilledRoundedRect(x, y, width, height, arcWidth, arcHeight, context);
    }

    public Rect toUnrounded() {
        return new Rect(x, y, width, height, context);
    }

    @Override
    public GraphicsContext context() {
        return context;
    }
}